ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -324,6 +329,52 @@ public void start() throws StoreException {
     }
   }
 
+  /**
+   * Find the last modified file in the folder
+   */
+  private File getLatestFilefromDir(String dirPath) {
+    File dir = new File(dirPath);
+    File[] files = dir.listFiles();
+    if (files == null || files.length == 0) {
+      return null;
+    }
+
+    File lastModifiedFile = files[0];
+    for (int i = 1; i < files.length; i++) {
+      if (lastModifiedFile.lastModified() < files[i].lastModified()) {
+        lastModifiedFile = files[i];
+      }
+    }
+    return lastModifiedFile;
+  }
+
+  /**
+   * Check if the store is stale.
+   * If yes, throw exception
+   */
+  private void checkIfStoreIsStale() throws StoreException {
+    File lastModifiedFile = getLatestFilefromDir(replicaId.getReplicaPath());
+
+    long lastModifiedTimeInMilliseconds = time.milliseconds();
+    if (lastModifiedFile != null) {
+      lastModifiedTimeInMilliseconds = lastModifiedFile.lastModified();
+    }
+    Date date = new Date(lastModifiedTimeInMilliseconds);
+    SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
+    String lastAliveTime = formatter.format(date);
+    logger.info(
+        "checkIfStoreIsStale " + replicaId.getReplicaPath() + " last modified file " + ((lastModifiedFile != null)
+            ? lastModifiedFile.getAbsolutePath() : "") + " last alive time " + lastAliveTime);
+
+    if (config.storeStaleTimeInDays != 0
+        && lastModifiedTimeInMilliseconds + TimeUnit.DAYS.toMillis(config.storeStaleTimeInDays) < time.milliseconds()) {
+      logger.error("checkIfStoreIsStale " + replicaId.getReplicaPath() + " is stale. ");
+      if (config.storeStaleBlockBootup) {
+        throw new StoreException("BlobStore is stale ", StoreErrorCodes.Initialization_Error);
```

Comment:
Also, can we add a GAUGE metric to show the number of stale blobstores.

Reply:
Added the Gauge and addressed all the comments. thanks,